### DISCUSSION TITLE
How to select vi keybindings?

### DISCUSSION THREAD

**Discussion by huyz** [nushell/nushell#8286] [Q&A] [answered]
I'm new.  First thing I try is `ls`.

Second thing I try is "let's find out the equivalent of `set -o vi` or `bindkey -v`".

I look at https://nushell.sh/book/line_editor.html , the most obvious place to figure out how to make my `Esc` key work to get into vi normal mode.  But it's not there.

I look at https://nushell.sh/book/coming_from_bash.html and it's not there either.

The `keybindings -h` command doesn't help.

I'm already frustrated and considering uninstalling as this already tells me something about the state of docs.  I guess I'll revisit in a few years.

**Comment by fdncred** [comment, marked as answer]
I think you have to configure nushell to use `edit_mode: vi` and then when specifying a keybinding you need to make sure `mode: [emacs, vi_normal, vi_insert]` matches whichever mode you want it to run in.

**Reply by huyz** [reply]
Thanks

**Reply by cassepipe** [reply]
This page suggests `edit_mode` instead : https://www.nushell.sh/book/line_editor.html

Is it outdated ?

**Reply by fdncred** [reply]
Yes, documentation is outdated. It should be `let-env` instead of `let`

**Comment by cassepipe** [comment]
I can't seem to make it work could you provide the one liner or  the minimal config I need to set it up ?

I have tried all the possible combinations that I know of. 
My config file is in nu's config path.

**Comment by cassepipe** [comment]
Ok so by default after first start `nu` gives you a default config file. Just edit the `edit_mode` field.

**Reply by Quoteme** [reply]
For anyone else wondering why even after changing `edit_mode` to `vi` nothing changes: Remember to restart nushell.
